platform/osgi/platform-osgi-condpermadmin/src/main/java/org/codice/ddf/condpermadmin/PermissionActivator.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.condpermadmin;
+
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import net.sourceforge.prograde.policy.SecurityActions;
+import net.sourceforge.prograde.policyparser.ParsedPermission;
+import net.sourceforge.prograde.policyparser.ParsedPolicy;
+import net.sourceforge.prograde.policyparser.ParsedPolicyEntry;
+import net.sourceforge.prograde.policyparser.ParsedPrincipal;
+import net.sourceforge.prograde.policyparser.Parser;
+import net.sourceforge.prograde.type.Priority;
+import org.apache.commons.lang3.text.StrSubstitutor;
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.condpermadmin.ConditionInfo;
+import org.osgi.service.condpermadmin.ConditionalPermissionAdmin;
+import org.osgi.service.condpermadmin.ConditionalPermissionInfo;
+import org.osgi.service.condpermadmin.ConditionalPermissionUpdate;
+import org.osgi.service.permissionadmin.PermissionInfo;
+import org.osgi.util.tracker.ServiceTracker;
+
+public class PermissionActivator implements BundleActivator {
+
+  private static final String BUNDLE_NAME_CONDITION =
+      "org.codice.ddf.condition.BundleNameCondition";
+
+  private static final String PRINCIPAL_CONDITION = "org.codice.ddf.condition.PrincipalCondition";
+
+  private static final String SIGNER_CONDITION = "org.codice.ddf.condition.SignerCondition";
+
+  private static final String OSGI_CODEBASE = "file:/";
```

Comment:
✏️ Back to my comment in the policy file, I suggest we change this to `osgi:`.

Reply:
I think it would be a little more work than this because the policy objects would also need to support the osgi stuff. I can make a ticket to look into this.